src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -2379,18 +2379,16 @@ static void read_action_locked(void* tp, grpc_error_handle error) {
   }
   std::swap(err, error);
   if (t->closed_with_error.ok()) {
-    size_t i = 0;
     grpc_error_handle errors[3] = {error, absl::OkStatus(), absl::OkStatus()};
-    for (; i < t->read_buffer.count && errors[1] == absl::OkStatus(); i++) {
+
+    for (size_t i = 0; i < t->read_buffer.count && errors[1] == absl::OkStatus(); i++) {
```

Comment:
do..while loop can be better here. Since the first iteration condition would be true anyways.

Reply:
Good idea.

Reply:
It's possible to have an empty buffer. See test errors. Please revert this bit, and just remove the unused loop in this PR.